stateless-asks/program/src/processor.rs

```diff
@@ -56,9 +68,80 @@ fn process_accept_offer(
     let taker_src_mint = next_account_info(account_info_iter)?;
     let transfer_authority = next_account_info(account_info_iter)?;
     let token_program_info = next_account_info(account_info_iter)?;
+    let mut system_program_info: Option<&AccountInfo> = None;
+    let is_native = *taker_src_mint.key == spl_token::native_mint::id();
+    if is_native {
+        assert_keys_equal(*taker_wallet.key, *taker_src_account.key)?;
+        assert_keys_equal(*maker_wallet.key, *maker_dst_account.key)?;
+        system_program_info = Some(next_account_info(account_info_iter)?);
+    }
+    let seeds = &[
+        b"stateless_offer",
+        maker_wallet.key.as_ref(),
+        maker_src_mint.key.as_ref(),
+        taker_src_mint.key.as_ref(),
+        &maker_size.to_le_bytes(),
+        &taker_size.to_le_bytes(),
+        &[bump_seed],
+    ];
+    let (maker_pay_size, taker_pay_size) = if has_metadata {
+        let metadata_info = next_account_info(account_info_iter)?;
+        let (maker_metadata_key, _) = Pubkey::find_program_address(
+            &[
+                b"metadata",
+                metaplex_token_metadata::id().as_ref(),
+                maker_src_mint.key.as_ref(),
+            ],
+            &metaplex_token_metadata::id(),
+        );
+        let (taker_metadata_key, _) = Pubkey::find_program_address(
+            &[
+                b"metadata",
+                metaplex_token_metadata::id().as_ref(),
+                taker_src_mint.key.as_ref(),
+            ],
+            &metaplex_token_metadata::id(),
+        );
+        if *metadata_info.key == maker_metadata_key {
+            msg!("Taker pays for fees");
+            let taker_remaining_size = pay_creator_fees(
+                account_info_iter,
+                metadata_info,
+                taker_src_account,
+                taker_wallet,
+                token_program_info,
+                system_program_info,
+                taker_src_mint,
+                taker_size,
+                is_native,
+                &[],
+            )?;
+            (maker_size, taker_remaining_size)
+        } else if *metadata_info.key == taker_metadata_key {
```

Comment:
Is there any demand to do an NFT for an NFT?

Reply:
oh really, how does that work? i thought both NFTs would operate totally independently, so any transfer of an NFT should pay out to its creators

Reply:
Yeah it totally should, but it's just a function of most NFT's having a supply of 1 and decimals of 0. When one of these is traded for another one, the creator fees will always round down to 0. This is unavoidable

Reply:
Oh of course, that makes perfect sense.  thanks for the explanation!

Reply:
I think an NFT for NFT feeless swap is a reasonable feature to keep in the protocol. Really no way to get around that